src/Struct/Exprs.hs

```diff
@@ -93,7 +94,8 @@ data Term =
   | TmAmb [Term] Type                             -- amb tm1 tm2 ... tmn : tp
   | TmFactor Double Term Type                     -- factor wt in tm : tp
   | TmProd AddMult [Arg]                          -- (tm1 : tp1, tm2 : tp2, ..., tmn : tpn) / <...>
-  | TmElimProd AddMult Term [Param] Term Type     -- let (x:X,y:Y,z:Z)/<...> = tm1 in tm2 : tp
+  | TmElimMultiplicative Term [Param] Term Type   -- let (x:X,y:Y,z:Z) = tm1 in tm2 : tp
+  | TmElimAdditive Term Int Int Param Term Type   -- let <_..,y:Y,_..> = tm1 in tm2 : tp
```

Comment:
Something possibly for a new issue: there's no place for the type of tm1 here, whereas generally the other cases of Term store the types of the subterms.

I think the direction to go in would be to reduce redundancy and not store the types of subterms (they can always be recovered using `typeof`), but was not sure if @HerbertMcSnout had a reason for designing the type like this.